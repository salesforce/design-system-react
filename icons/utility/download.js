/* Copyright (c) 2015-present, salesforce.com, inc. All rights reserved */
/* Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license */


export default {"viewBox":"0 0 24 24","path":{"d":"M22.4 14.3H21c-.4 0-.7.3-.7.7v4.6c0 .4-.3.7-.7.7H4.4c-.4 0-.7-.3-.7-.7V15c0-.4-.3-.7-.7-.7H1.6c-.4 0-.7.3-.7.7v6.2c0 1 .9 1.9 1.9 1.9h18.4c1 0 1.9-.9 1.9-1.9V15c0-.4-.3-.7-.7-.7zm-10.9 3.1c.3.2.7.2 1 0l6.2-6.3c.3-.3.3-.7 0-.9l-.9-1c-.3-.3-.7-.3-1 0l-2.6 2.6c-.3.2-.8.1-.8-.4V1.6c0-.4-.4-.7-.7-.7h-1.4c-.4 0-.7.3-.7.7v9.8c0 .4-.5.6-.8.3L7.2 9.1c-.2-.2-.6-.2-.9 0l-1 1.1c-.3.2-.3.6 0 .9l6.2 6.3z"}};
