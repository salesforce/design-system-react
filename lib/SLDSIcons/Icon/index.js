/*
Copyright (c) 2015, salesforce.com, inc. All rights reserved.

Redistribution and use in source and binary forms, with or without modification, are permitted provided that the following conditions are met:
Redistributions of source code must retain the above copyright notice, this list of conditions and the following disclaimer.
Redistributions in binary form must reproduce the above copyright notice, this list of conditions and the following disclaimer in the documentation and/or other materials provided with the distribution.
Neither the name of salesforce.com, inc. nor the names of its contributors may be used to endorse or promote products derived from this software without specific prior written permission.

THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
*/

'use strict';

var _createClass = (function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ('value' in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; })();

var _get = function get(_x, _x2, _x3) { var _again = true; _function: while (_again) { var object = _x, property = _x2, receiver = _x3; _again = false; if (object === null) object = Function.prototype; var desc = Object.getOwnPropertyDescriptor(object, property); if (desc === undefined) { var parent = Object.getPrototypeOf(object); if (parent === null) { return undefined; } else { _x = parent; _x2 = property; _x3 = receiver; _again = true; desc = parent = undefined; continue _function; } } else if ('value' in desc) { return desc.value; } else { var getter = desc.get; if (getter === undefined) { return undefined; } return getter.call(receiver); } } };

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { 'default': obj }; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError('Cannot call a class as a function'); } }

function _inherits(subClass, superClass) { if (typeof superClass !== 'function' && superClass !== null) { throw new TypeError('Super expression must either be null or a function, not ' + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _SLDSUtilityIcon = require('../../SLDSUtilityIcon');

var _SLDSUtilityIcon2 = _interopRequireDefault(_SLDSUtilityIcon);

var classNames = require("classnames");

var displayName = "Icon";
var propTypes = {
  /**
   * Text that is visually hidden but read aloud by screenreaders to tell the user what the icon means.
   * Naked icons must have assistive text, however, if you also have visible descriptive text with the icon,
   * declare this prop as <code>assistiveText=""</code>.
   */
  assistiveText: _react2['default'].PropTypes.string,
  category: _react2['default'].PropTypes.oneOf(["action", "custom", "doctype", "standard", "utility"]).isRequired,
  /**
   * css classes that are applied to the svg
   */
  className: _react2['default'].PropTypes.string,
  /**
   * Name of the icon. Visit <a href="http://www.lightningdesignsystem.com/resources/icons">SLDS Icons</a> to reference icon names.
   */
  name: _react2['default'].PropTypes.string.isRequired,
  /**
   * If omitted, icon size is medium.
   */
  size: _react2['default'].PropTypes.oneOf(["x-small", "small", "large"])
};
var defaultProps = {
  category: 'standard'
};

/**
 * The Icon component should be used for icons only. For icons that are buttons, use the Button component. <br />
 * The icon color is white by default. Add the class, <code>slds-icon-text-default</code>, to create a text-colored fill color for utility icons. <br />
 * For more details, please reference <a href="http://www.lightningdesignsystem.com/components/icons">SLDS Icons</a>.
 */

var Icon = (function (_React$Component) {
  _inherits(Icon, _React$Component);

  function Icon(props) {
    _classCallCheck(this, Icon);

    _get(Object.getPrototypeOf(Icon.prototype), 'constructor', this).call(this, props);
    this.state = {};
  }

  _createClass(Icon, [{
    key: 'getContainerClassName',
    value: function getContainerClassName() {
      var _classNames;

      var name = this.props.name ? this.props.name.replace(/_/g, '-') : '';
      var renderName = this.props.category === "action";

      return classNames((_classNames = {}, _defineProperty(_classNames, "slds-icon__container", this.props.category !== "utility"), _defineProperty(_classNames, 'slds-icon-' + this.props.category + '-' + name, renderName), _classNames));
    }
  }, {
    key: 'getClassName',
    value: function getClassName() {
      var _classNames2;

      var name = this.props.name ? this.props.name.replace(/_/g, '-') : '';
      var customName = this.props.name ? this.props.name.replace("custom", "custom-") : null;

      return classNames(this.props.className, "slds-icon", (_classNames2 = {}, _defineProperty(_classNames2, 'slds-icon--' + this.props.size, this.props.size), _defineProperty(_classNames2, 'slds-icon-' + customName, this.props.category === "custom"), _defineProperty(_classNames2, 'slds-icon-' + this.props.category + '-' + name, this.props.category === "standard"), _classNames2));
    }
  }, {
    key: 'render',
    value: function render() {
      var label = null;

      if (this.props.assistiveText) {
        label = _react2['default'].createElement(
          'span',
          { className: 'slds-assistive-text' },
          this.props.assistiveText
        );
      }
      return _react2['default'].createElement(
        'span',
        { className: this.getContainerClassName() },
        label,
        _react2['default'].createElement(_SLDSUtilityIcon2['default'], {
          'aria-hidden': 'true',
          category: this.props.category,
          className: this.getClassName(),
          name: this.props.name
        })
      );
    }
  }]);

  return Icon;
})(_react2['default'].Component);

Icon.displayName = displayName;
Icon.propTypes = propTypes;
Icon.defaultProps = defaultProps;

module.exports = Icon;