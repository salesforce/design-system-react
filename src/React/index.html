<!doctype html public "library">
<meta charset=utf-8/>
<head>
	<title>Selectlist</title>
	<link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.5/css/bootstrap.min.css">
</head>
<body>
<script src="/dist/react/build.js"></script>
</body>
</html>