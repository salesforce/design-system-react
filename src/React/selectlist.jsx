//test

var React = require('react');
import {SelectlistCore} from "../SelectlistCore";
var DropdownButton = require('react-bootstrap/lib/DropdownButton');
var MenuItem = require('react-bootstrap/lib/MenuItem');

var DATA = {
	selected: 'None selected',
	id: 'selectlist1',
	items: [
		{ id: 0, name: 'tacos', type: 'mexican' },
		{ id: 1, name: 'burrito', type: 'mexican' },
		{ id: 2, name: 'tostada', type: 'mexican' },
		{ id: 3, name: 'hush puppies', type: 'southern' }
	]
};

var menuItemSelected = (eventKey, href, target)=> {
	DATA.selected = DATA.items[eventKey].name;
};
//Selectlist.handleMenuItemClicked
var menuItems = DATA.items.map( (menuItem) => {
	return <MenuItem eventKey={menuItem.id}>{menuItem.name}</MenuItem>;
});

var Selectlist = React.createClass({
	getInitialState: function() {
		return {
			selectedKey: undefined
		};
	},

	handleMenuItemClicked: function(eventKey, href, target) {
		this.setState({
			selectedKey: eventKey
		});
	},

	menuItems: function() {
		return DATA.items.map( (menuItem) => {
			return <MenuItem eventKey={menuItem.id} onSelect={this.handleMenuItemClicked}>{menuItem.name}</MenuItem>;
		});
	},

	selected: function() {
		return typeof(this.state.selectedKey) === 'undefined' ? 'None Selected' : DATA.items[this.state.selectedKey].name;
	},

	render () {
		return (
			<div>
				<h1>SelectList</h1>
				<ul>{this.menuItems}</ul>
				<DropdownButton title={this.selected()} key={DATA.id}>{this.menuItems()}</DropdownButton>
			</div>
		);
	}
});

React.render(<Selectlist/>, document.body, () => {});