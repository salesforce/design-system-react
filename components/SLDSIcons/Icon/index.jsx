/*
Copyright (c) 2015, salesforce.com, inc. All rights reserved.

Redistribution and use in source and binary forms, with or without modification, are permitted provided that the following conditions are met:
Redistributions of source code must retain the above copyright notice, this list of conditions and the following disclaimer.
Redistributions in binary form must reproduce the above copyright notice, this list of conditions and the following disclaimer in the documentation and/or other materials provided with the distribution.
Neither the name of salesforce.com, inc. nor the names of its contributors may be used to endorse or promote products derived from this software without specific prior written permission.

THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
*/

import React from 'react';
import SLDSUtilityIcon from '../../SLDSUtilityIcon';
const classNames = require("classnames");

const displayName = "Icon";
const propTypes = {
  /**
   * Text that is visually hidden but read aloud by screenreaders to tell the user what the icon means.
   * Naked icons must have assistive text, however, if you also have visible descriptive text with the icon,
   * declare this prop as <code>assistiveText=""</code>.
   */
  assistiveText: React.PropTypes.string,
  category: React.PropTypes.oneOf(["action", "custom", "doctype", "standard", "utility"]).isRequired,
  /**
   * css classes that are applied to the svg
   */
  className: React.PropTypes.string,
  /**
   * Name of the icon. Visit <a href="http://www.lightningdesignsystem.com/resources/icons">SLDS Icons</a> to reference icon names.
   */
  name: React.PropTypes.string.isRequired,
  /**
   * If omitted, icon size is medium.
   */
  size: React.PropTypes.oneOf(["x-small", "small", "large"]),
};
const defaultProps = {
  category: 'standard',
};

/**
 * The Icon component should be used for icons only. For icons that are buttons, use the Button component. <br />
 * The icon color is white by default. Add the class, <code>slds-icon-text-default</code>, to create a text-colored fill color for utility icons. <br />
 * For more details, please reference <a href="http://www.lightningdesignsystem.com/components/icons">SLDS Icons</a>.
 */
class Icon extends React.Component {

  constructor(props) {
    super(props);
    this.state = {};
  }

  getContainerClassName() {
    const name = this.props.name ? this.props.name.replace(/_/g,'-') : '';
    const renderName = (this.props.category === "action");

    return classNames({
      ["slds-icon__container"]: this.props.category !== "utility",
      [`slds-icon-${this.props.category}-${name}`]: renderName,
    })
  }

  getClassName() {
    const name = this.props.name ? this.props.name.replace(/_/g,'-') : '';
    const customName = this.props.name ? this.props.name.replace("custom", "custom-"): null;

    return classNames(this.props.className, "slds-icon", {
      [`slds-icon--${this.props.size}`]: this.props.size,
      [`slds-icon-${customName}`]: this.props.category === "custom",
      [`slds-icon-${this.props.category}-${name}`]: this.props.category === "standard",
    });
  }

  render() {
    let label = null;

    if(this.props.assistiveText) {
      label = <span className="slds-assistive-text">{this.props.assistiveText}</span>;
    }
    return (
      <span className={this.getContainerClassName()}>
      {label}
      <SLDSUtilityIcon
        aria-hidden='true'
        category={this.props.category}
        className={this.getClassName()}
        name={this.props.name}
        />
      </span>
    )
  }
}

Icon.displayName = displayName;
Icon.propTypes = propTypes;
Icon.defaultProps = defaultProps;

module.exports = Icon;

