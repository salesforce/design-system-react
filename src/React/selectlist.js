// SELECTLIST CONTROL

// core
import {Landmark} from '../Landmark';
import {SelectlistCore} from '../SelectlistCore';

// framework specific
import React from 'react';
import DropdownButton from 'react-bootstrap/lib/DropdownButton';
import MenuItem from 'react-bootstrap/lib/MenuItem';

export var Selectlist = React.createClass(Object.assign({}, SelectlistCore, {	
	propTypes: {
		disabled: React.PropTypes.bool,
		selection: React.PropTypes.number,
		collection: React.PropTypes.array
	},
	
	getInitialState () {
		return this.__getInitialState();
	},
		
	menuItems () {
		return this.props.collection.map((menuItem) => {
			return <MenuItem eventKey={menuItem.id} onSelect={this.handleMenuItemClicked}>
					{menuItem.name}
				</MenuItem>;
		});
	},
	
	componentWillReceiveProps(nextProps) {
		this.__initializeOptions(nextProps);
	},
	
	render () {
		var selection = this.getSelection();
		
		return (
			<DropdownButton className={this.cssClasses.CONTROL} disabled={this.state.disabled} title={selection ? selection.name : 'None selected'} key={this.props.id}>{this.menuItems()}</DropdownButton>
		);
	},
	
	componentWillMount () {
		this.elements = {
			wrapper: {
				toggleClass: function (cssClass, state) {
					Landmark.log(cssClass, state);
				}
			}
		}
		
		this.__constructor(this.props);
	},
	
	handleMenuItemClicked (eventKey, href, target) {
		this.setSelectionByKey('id', eventKey);
	}
}));