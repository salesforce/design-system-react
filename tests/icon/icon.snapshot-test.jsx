/* eslint-env jest */
import React from 'react';
import renderer from 'react-test-renderer';
import { renderMarkup } from '../snapshot-helpers';

import Icon from '../../components/icon';
import download from '../../icons/utility/download';

import Standard from '../../examples/icon/standard';
import Utility from '../../examples/icon/utility';
import Action from '../../examples/icon/action';
import Doctype from '../../examples/icon/doctype';
import Custom from '../../examples/icon/custom';
import ExternalPath from '../../examples/icon/external-path';

import ColorBase from '../../examples/icon/color-base';
import ColorDefault from '../../examples/icon/color-default';
import ColorError from '../../examples/icon/color-error';
import ColorWarning from '../../examples/icon/color-warning';

import SizesExtraSmall from '../../examples/icon/sizes-extra-small';
import SizesSmall from '../../examples/icon/sizes-small';
import SizesMedium from '../../examples/icon/sizes-medium';
import SizesLarge from '../../examples/icon/sizes-large';

test('Icon Category Standard HTML Snapshot', () => {
	expect(renderMarkup(Standard)).toMatchSnapshot();
});

test('Icon Category Utility HTML Snapshot', () => {
	expect(renderMarkup(Utility)).toMatchSnapshot();
});

test('Icon Category Action HTML Snapshot', () => {
	expect(renderMarkup(Action)).toMatchSnapshot();
});

test('Icon Category Doctype HTML Snapshot', () => {
	expect(renderMarkup(Doctype)).toMatchSnapshot();
});

test('Icon Category Custom HTML Snapshot', () => {
	expect(renderMarkup(Custom)).toMatchSnapshot();
});

test('Icon Category External Path HTML Snapshot', () => {
	expect(renderMarkup(ExternalPath)).toMatchSnapshot();
});

test('Icon Size X-Small HTML Snapshot', () => {
	expect(renderMarkup(SizesExtraSmall)).toMatchSnapshot();
});

test('Icon Size Small HTML Snapshot', () => {
	expect(renderMarkup(SizesSmall)).toMatchSnapshot();
});

test('Icon Size Medium HTML Snapshot', () => {
	expect(renderMarkup(SizesMedium)).toMatchSnapshot();
});

test('Icon Size Large HTML Snapshot', () => {
	expect(renderMarkup(SizesLarge)).toMatchSnapshot();
});

