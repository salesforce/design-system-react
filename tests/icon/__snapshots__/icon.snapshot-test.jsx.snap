// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon Category Action HTML Snapshot 1`] = `"<span class=\\"slds-icon_container slds-icon_container--circle slds-icon-action-description\\"><svg aria-hidden=\\"true\\" class=\\"slds-icon slds-icon--small\\"><use xlink:href=\\"/assets/icons/action-sprite/svg/symbols.svg#description\\"></use></svg><span class=\\"slds-assistive-text\\">Description of icon</span></span>"`;

exports[`Icon Category Custom HTML Snapshot 1`] = `"<span class=\\"slds-icon_container slds-icon-custom-custom5\\" title=\\"description of icon when needed\\"><svg aria-hidden=\\"true\\" class=\\"slds-icon\\"><use xlink:href=\\"/assets/icons/custom-sprite/svg/symbols.svg#custom5\\"></use></svg><span class=\\"slds-assistive-text\\">Description of icon</span></span>"`;

exports[`Icon Category Doctype HTML Snapshot 1`] = `"<span class=\\"slds-icon_container\\"><svg aria-hidden=\\"true\\" class=\\"slds-icon\\"><use xlink:href=\\"/assets/icons/doctype-sprite/svg/symbols.svg#xml\\"></use></svg><span class=\\"slds-assistive-text\\">Description of icon</span></span>"`;

exports[`Icon Category External Path HTML Snapshot 1`] = `"<span class=\\"slds-icon_container\\"><svg aria-hidden=\\"true\\" class=\\"slds-icon slds-icon-text-default\\"><use xlink:href=\\"/assets/icons/utility-sprite/svg/symbols.svg#announcement\\"></use></svg><span class=\\"slds-assistive-text\\">We got news!</span></span>"`;

exports[`Icon Category Standard HTML Snapshot 1`] = `"<span class=\\"slds-icon_container slds-icon-standard-account\\" title=\\"description of icon\\"><svg aria-hidden=\\"true\\" class=\\"slds-icon\\"><use xlink:href=\\"/assets/icons/standard-sprite/svg/symbols.svg#account\\"></use></svg><span class=\\"slds-assistive-text\\">Description of icon</span></span>"`;

exports[`Icon Category Utility HTML Snapshot 1`] = `"<span title=\\"description of icon when needed\\"><svg aria-hidden=\\"true\\" class=\\"slds-icon slds-icon--small slds-icon-text-default\\"><use xlink:href=\\"/assets/icons/utility-sprite/svg/symbols.svg#announcement\\"></use></svg><span class=\\"slds-assistive-text\\">Description of icon</span></span>"`;

exports[`Icon Size Large HTML Snapshot 1`] = `"<span class=\\"slds-icon_container slds-icon-standard-case\\"><svg aria-hidden=\\"true\\" class=\\"slds-icon slds-icon--large\\"><use xlink:href=\\"/assets/icons/standard-sprite/svg/symbols.svg#case\\"></use></svg><span class=\\"slds-assistive-text\\">Description of icon</span></span>"`;

exports[`Icon Size Medium HTML Snapshot 1`] = `"<span class=\\"slds-icon_container slds-icon-standard-case\\"><svg aria-hidden=\\"true\\" class=\\"slds-icon\\"><use xlink:href=\\"/assets/icons/standard-sprite/svg/symbols.svg#case\\"></use></svg><span class=\\"slds-assistive-text\\">Description of icon</span></span>"`;

exports[`Icon Size Small HTML Snapshot 1`] = `"<span class=\\"slds-icon_container slds-icon-standard-case\\"><svg aria-hidden=\\"true\\" class=\\"slds-icon slds-icon--small\\"><use xlink:href=\\"/assets/icons/standard-sprite/svg/symbols.svg#case\\"></use></svg><span class=\\"slds-assistive-text\\">Description of icon</span></span>"`;

exports[`Icon Size X-Small HTML Snapshot 1`] = `"<span><svg aria-hidden=\\"true\\" class=\\"slds-icon slds-icon--x-small slds-icon-text-default\\"><use xlink:href=\\"/assets/icons/utility-sprite/svg/symbols.svg#warning\\"></use></svg><span class=\\"slds-assistive-text\\">Description of icon</span></span>"`;
